server/queue/queue-file/src/test/java/org/apache/james/queue/file/FileMailQueueTest.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Override
+    public MailQueue getMailQueue() {
+        return mailQueue;
+    }
+
+    @Override
+    public ManageableMailQueue getManageableMailQueue() {
+        return mailQueue;
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2300 No Order")
+    public void browseShouldReturnElementsInOrder() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2300 No Order")
+    public void flushShouldPreserveBrowseOrder() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentClearShouldNotAlterBrowsingWhenDequeueWhileIterating() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentClearShouldNotAlterBrowsingWhenDequeue() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentDequeueShouldNotAlterBrowsing() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentEnqueueShouldNotAlterBrowsing() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentDequeueShouldNotAlterBrowsingWhenDequeueWhileIterating() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentEnqueueShouldNotAlterBrowsingWhenDequeueWhileIterating() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentRemoveShouldNotAlterBrowsingWhenDequeueWhileIterating() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentFlushShouldNotAlterBrowsingWhenDequeueWhileIterating() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentFlushShouldNotAlterBrowsing() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2299 No snapshot isolation")
+    public void concurrentRemoveShouldNotAlterBrowsingWhenDequeue() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2298 Not supported yet")
+    public void removeBySenderShouldRemoveSpecificEmail() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2298 Not supported yet")
+    public void removeByNameShouldRemoveSpecificEmail() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2298 Not supported yet")
+    public void removeByRecipientShouldRemoveSpecificEmail() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2298 Not supported yet")
+    public void removeByRecipientShouldNotFailWhenQueueIsEmpty() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2298 Not supported yet")
+    public void removeBySenderShouldNotFailWhenQueueIsEmpty() {
+
+    }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2298 Not supported yet")
+    public void removeByRecipientShouldRemoveSpecificEmailWhenMultipleRecipients() {
```

Comment:
maybe all those @ignore means there are several levels of "manageable", don't you think ?

Reply:
Don't you fear that, by making it too composable, we end up with a dependancy hell?

I can introduce a **BasicManageableMailQueue** without remove, if you wish.

Reply:
We discussed that privatly with @mbaechler  and came to the conclusion that:

 - `We will not support FileMailQueue with WebAdmin`
 - `We don't want experimental components to redefine otherwise consistant Interfaces and **thus impact our code**`

As a result, we will keep current solution.